Fix attachment filename generation for URLs with query parameters

obenland opened the pull request:
## Proposed changes:
* Strip query parameters from URLs before generating attachment filenames
* Prevents "Filename too long" filesystem errors
* Fixes issues with Instagram CDN URLs and other services that use long query strings

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

1. Process an ActivityPub activity with an attachment URL that has query parameters (e.g., Instagram CDN URL)
2. Verify the file is saved successfully without errors
3. Check that the filename is clean without query parameters

**Before:** 
```
PHP Warning: copy(.../582455906_17930138757114605_3007218277985087381_n.jpgstpdst-jpg_e35_tt6_nc_cat101ccb7-5_nc_sid18de74efgeyJlZmdfdGFnIjoiRkVFRC5iZXN0X2ltYWdlX3VybGdlbi5DMyJ9_nc_ohcne9zSzVebdsQ7kNvwGMeb6Q_nc_ocAdmzsGgwxVX3rDrTgZriddqj37P8s3eVEtAa8tNuy5DnRzJUT7nFkflSQgNVztbdQCIpM6rhfFycR1UGy1LYI2oP_nc_adz-m_nc_cid0_nc_zt23_nc_htscontent.cdninstagram.com_nc_gideIO13gOfATKzYZ-ERRIiugoh00_Afg0LbbH0emgIK49rQawuJMuNFn3fJZCkBf5zfuflpG90Qoe6922C08B): Failed to open stream: Filename too long
```

**After:** 
File saved successfully as `582455906_17930138757114605_3007218277985087381_n.jpg`

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [x] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Fix "Filename too long" errors when downloading attachments from URLs with query parameters (e.g., Instagram CDN URLs).

</details>

obenland committed:
Fix attachment filename generation for URLs with query parameters

Strip query parameters from URLs before generating filenames to prevent
'Filename too long' errors. This is particularly important for Instagram
and other CDN URLs that include long query strings with cache-busting
parameters.

Without this fix, URLs like:
  https://example.com/image.jpg?stp=dst-jpg_e35&nc_cat=101&...
Would generate filenames like:
  image.jpgstp=dst-jpg_e35_nc_cat=101_... (400+ characters)

Now correctly generates:
  image.jpg

Fixes in both save_attachment() and save_file() methods.

matticbot committed:
Add changelog

obenland commented on includes/class-attachments.php:
```diff
@@ -446,7 +446,7 @@ private static function save_attachment( $attachment_data, $post_id, $author_id
 		// Prepare file array for WordPress.
 		$file_array = array(
-			'name'     => \basename( $attachment_data['url'] ),
+			'name'     => \basename( \strtok( $attachment_data['url'], '?' ) ),
```
it's fine

obenland committed:
Add unit tests for query parameter stripping in attachment filenames

Tests verify that:
- save_attachment() strips query parameters from attachment URLs
- save_file() strips query parameters from direct file storage URLs
- Filenames are clean and don't contain ? or query string components

This ensures the fix for 'Filename too long' errors is properly tested.

obenland committed:
Fix test failures and lint issues

- Remove unused variable to fix phpcs warning
- Use assertStringEndsWith instead of assertStringContainsString for file extension checks

pfefferle commented on includes/class-attachments.php:
```diff
@@ -446,7 +446,7 @@ private static function save_attachment( $attachment_data, $post_id, $author_id
 		// Prepare file array for WordPress.
 		$file_array = array(
-			'name'     => \basename( $attachment_data['url'] ),
+			'name'     => \basename( \strtok( $attachment_data['url'], '?' ) ),
```
I am not sure how likely this is, but should we also strip anchors (`example.com#abc`)?

obenland replied:
Fixed in [7ff2eb1](https://github.com/Automattic/wordpress-activitypub/pull/2499/commits/7ff2eb1666f81f7e2d33450cccc66e28cad6a461).

obenland committed:
Use wp_parse_url to strip both query params and fragments

Changed from strtok() to wp_parse_url() to properly handle URL fragments
(anchors with #) in addition to query parameters. This matches WordPress
core's approach in download_url() and ensures filenames never contain
fragment identifiers.

obenland committed:
Merge branch 'trunk' into fix/attachment-filename-query-params
